src/Uno.UI/Extensions/TimeSpanExtensions.cs

```diff
@@ -75,5 +75,26 @@ internal static TimeSpan RoundToMinuteInterval(this TimeSpan time, int interval)
 
 			return time;
 		}
+
+		/// <summary>
+		/// Normalize TimeSpan between 0 and 24h
+		/// </summary>
+		/// <returns>Normalized TimeSpan</returns>
+		internal static TimeSpan NormalizeToDay(this TimeSpan value)
+		{
+			var day = TimeSpan.FromDays(1);
+
+			while (value < TimeSpan.Zero)
```

Comment:
`while` loops are unnecessary here, you do it with modulo:

    internal static TimeSpan NormalizeToDay2(TimeSpan value)
    {
        return new TimeSpan(Mod(value.Ticks, TimeSpan.TicksPerDay));
    
        long Mod(long x, long m)
        {
            long r = x % m;
            return r < 0 ? r + m : r;
        }
    }

Reply:
I thought about it, time complexity is definitely less, but readability is worse.
For this particular PR `while` loop won't be executed more than once. 
Do you think it's still worth to use modulo here?

Reply:
Modulo is more readable than two while loops.

As for the edge case above, you can make an early exit with:
```
if (value.Ticks == TimeSpan.TicksPerDay) return value;
```